File: php-pr/51/php/code-quality-performance/ArraySearcher.php
<?php

class ArraySearcher
{
    private $numbers;

    public function __construct()
    {
        $this->numbers = [];
        for ($i = 0; $i < 10000; $i++) {
            $this->numbers[] = $i;
        }
    }

    public function findNumber($target)
    {
        // {fact rule=code-quality-performance@v1.0 defects=1}
        foreach ($this->numbers as $number) {
            if ($number === $target) {
                return true;
            }
        }
        return false;
        // {/fact}
    }
}

File: kotlin-pr/50/kotlin/code-quality-performance/ArraySearcher.kt
class ArraySearcher {
    private val numbers = (0 until 10000).toList()

    fun findNumber(target: Int): Boolean {
        // {fact rule=code-quality-performance@v1.0 defects=1}
        numbers.forEach { number ->
            if (number == target) return true
        }
        return false
        // {/fact}
    }
}